### ISSUE TITLE
Change "Starting session" to look like the vanilla loading terrain screen

### ISSUE THREAD

**Issue by ghost** [BumbleSoftware/Fastload#68]
Those custom loading screens are a bit scary, like seriously some people might get confused too.

**Comment by Talbe1** [comment]
Why are they a bit scary?
Are you talking about Photosensitive epilepsy?

I'm **NOT** the developer himself by any means (duh), but in the latest version of the mod (3.2.0) the 441 loading engine was removed entirely (as it was mentioned in the changelog) and I **assume** that it was replaced with a much faster solution.

Also, if you didn't know, there's a bug when joining online servers that causes you to be unable to "fully" join them (issue #63).

This is just my opinion, based on the minimal testing of the new version that I did,
since the new solution reduces world loading time **SO** much quicker, there might not be a point to try and
recreate or make a replica of the vanilla loading terrain screen.

**Comment by StockiesLad** [comment]
> Why are they a bit scary? Are you talking about Photosensitive epilepsy?
> 
> I'm **NOT** the developer himself by any means (duh), but in the latest version of the mod (3.2.0) the 441 loading engine was removed entirely (as it was mentioned in the changelog) and I **assume** that it was replaced with a much faster solution.
> 
> Also, if you didn't know, there's a bug when joining online servers that causes you to be unable to "fully" join them (issue #63).
> 
> This is just my opinion, based on the minimal testing of the new version that I did, since the new solution reduces world loading time **SO** much quicker, there might not be a point to try and recreate or make a replica of the vanilla loading terrain screen.

Basically what this guy said. There is no point having a screen where it doesn't show information. 441 is now only set to load 1 chunk (the spawn one). There is no point showing that complex (and very laggy screen actually) for 1 chunk.

**Comment by StockiesLad** [comment]
> Those custom loading screens are a bit scary, like seriously some people might get confused too.

Starting session is the most accurate text I could think of. It applies universally to Boot, Loading, Servers & Singleplayer.

**Comment by ghost** [comment]
Starting session shows no information either. This is the kind of breakage to vanilla parity people are going to see as weird.

**Comment by StockiesLad** [comment]
> Starting session shows no information either. This is the kind of breakage to vanilla parity people are going to see as weird.

There isn't any information to be presented. It's literally starting the session. What kind of information do you want to see?

**Comment by Kichura** [comment]
tbh the "starting session" is not worth having as it adds extra delay towards loading, in previous versions of fastload itself *(which is 3.1.1 in this case)* - it allows the player to see the progress system properly in order to indicate if the world has actually loaded properly or not, but with the text only instead; this makes it nearly impossible to tell if world successfully makes any loading progress or not, hence people will complain about it here.

In short, i fail to personally see this as being "laggy" whatsoever.

**Comment by StockiesLad** [comment]
> tbh the "starting session" is not worth having as it adds extra delay towards loading, in previous versions of fastload itself _(which is 3.1.1 in this case)_ - it allows the player to see the progress system properly in order to indicate if the world has actually loaded properly or not, but with the text only instead; this makes it nearly impossible to tell if world successfully makes any loading progress or not, hence people will complain about it here.
> 
> In short, i fail to personally see this as being "laggy" whatsoever.

Starting session is not a different screen. I can probably add some text changes that will display progress as such that is has some parity of vanilla. There is no such thing as just "removing it" because that and building terrain screen are the same screen.

**Comment by StockiesLad** [comment]
> > tbh the "starting session" is not worth having as it adds extra delay towards loading, in previous versions of fastload itself _(which is 3.1.1 in this case)_ - it allows the player to see the progress system properly in order to indicate if the world has actually loaded properly or not, but with the text only instead; this makes it nearly impossible to tell if world successfully makes any loading progress or not, hence people will complain about it here.
> > In short, i fail to personally see this as being "laggy" whatsoever.
> 
> Starting session is not a different screen. I can probably add some text changes that will display progress as such that is has some parity of vanilla. There is no such thing as just "removing it" because that and building terrain screen are the same screen.

I'm thinking something along the lines of "Starting Server", "Joining World"

**Comment by Kichura** [comment]
> > > tbh the "starting session" is not worth having as it adds extra delay towards loading, in previous versions of fastload itself _(which is 3.1.1 in this case)_ - it allows the player to see the progress system properly in order to indicate if the world has actually loaded properly or not, but with the text only instead; this makes it nearly impossible to tell if world successfully makes any loading progress or not, hence people will complain about it here.
> > > In short, i fail to personally see this as being "laggy" whatsoever.
> > 
> > 
> > Starting session is not a different screen. I can probably add some text changes that will display progress as such that is has some parity of vanilla. There is no such thing as just "removing it" because that and building terrain screen are the same screen.
> 
> I'm thinking something along the lines of "Starting Server", "Joining World"

"Joining world" would make sense and atleast a progress bar should be displayed of some kind because then it would make a bit more sense on what is happening exactly.

**Comment by Felix14-v2** [comment]
I also don't like the fact that FastLoad changes the appearance of vanilla by default. I also understand that disabling the chunk loading graph is a step towards increasing the loading speed. But imo this should have been made optional and disabled by default, so as not to break the game appearance without asking the player's consent.

Also, this causes some compatibility issues at the moment:

https://user-images.githubusercontent.com/75726196/230788751-afffe580-a93f-4268-b080-6a738ed25293.mp4

I have not yet figured out the mod that causes such a conflict, but I'll definitely do it later.

**Comment by StockiesLad** [comment]
> I also don't like the fact that FastLoad changes the appearance of vanilla by default. I also understand that disabling the chunk loading graph is a step towards increasing the loading speed. But imo this should have been made optional and disabled by default, so as not to break the game appearance without asking the player's consent.
> 
> Also, this causes some compatibility issues at the moment:
> 
> I have not yet figured out the mod that causes such a conflict, but I'll definitely do it later.

I can make it configurable but I am not keen on disabling it by default. It is pointless to waste resources on rendering a chunkmap that has no valuable information. If one wants to switch it on, on their own accord, I am happy to provide the ability to do so.

**Comment by StockiesLad** [comment]
What will happen is that a chunk map will appear, get a green dot in the middle, then disappear...

Furthermore, on the basis of player consent, if mods asked the player for consent for any game changing thing, people will be sitting there for like 15 minutes clicking "yes". This mod has it's set of desired features, with their corresponding config so one is able to customise away from the **intended experience** to their own desires.

**Comment by Felix14-v2** [comment]
> Also, this causes some compatibility issues at the moment

Wasn't an issue in 2.7, btw

**Comment by StockiesLad** [comment]
> > Also, this causes some compatibility issues at the moment
> 
> Wasn't an issue in 2.7, btw

Yes it shouldn't because 2.7 & 3.3.1 are very, very different codewise.

**Comment by StockiesLad** [comment]
Ok I have been testing a lot. Letting the chunk map return makes absolutely 0 sense because you won't see progress anyways. I tried to rectify this by added text stages for BuildingTerrainScreen but since there is basically only 1 stage: session starting, it makes no sense to replace this. You won't be missing out on any progress at all. 

Let me break it down, there are 3 stages. World preparation, 441, Server/Session init, rendering/joining world.

I don't touch world preparation and 441 is removed, which only leaves us with Server/Session init. There is no way to just break this down into tiny stages so that you guys can see the progress, I know because I tried and failed. In order to successfully do this, I have to make a packet system for fastload, which is not going to happen.  Therefore, fastload can only present the three stages that there are.  If you don't believe me, compile from branch "feat/unfeature" and you can see for yourself, the 441 chunkmap does nothing. If you really, really want some sort of objective measurement of progress, fastload is open source and you are free to PR it. If it's good I'll accept it and probably give you some cut of the money fastload makes. Otherwise, this case is closed.
